### PR TITLE
Update LocalSettings.php

### PR DESCRIPTION
**Pull request by bonnedav**
Add the researcher group I need for testing.
Dedupe permissions.
Fixed some groups and permissions.
Add translateadmin and confirmed groups which I think would be useful.

### CODE REVIEW

**Comment by Psl85** [comment]
Please change also "default skin" from refreshed to vector please :)

**Comment by bonnedav** [comment]
@Psl85 Do that in you own PR. This is for user rights improvements. Do you have any comments on the current changes or suggestions for other ones?

**Comment by The-Voidwalker** [comment]
Ideally we are not going to set settings to false just because they are redundant

File: LocalSettings.php [hunk 1]
```diff
@@ -305,34 +327,56 @@
...
+$wgGroupPermissions['editor']['editsemiprotected'] = true;
+$wgGroupPermissions['editor']['unreviewedpages'] = true;
+$wgGroupPermissions['autopatrol']['autopatrol'] = true;
+$wgAddGroups['sysop'] = array( 'editor', 'reviewer', 'autoreview', 'autopatrol', 'patroller', 'translateadmin', 'confirmed', 'researcher', 'chatmod', 'blockedfromchat' );
+$wgRemoveGroups['sysop'] = array( 'editor', 'reviewer', 'autoreview', 'autopatrol', 'patroller', 'translateadmin', 'confirmed', 'researcher', 'chatmod', 'blockedfromchat' );
+$wgAddGroups['steward'] = array( 'sysop', 'bureaucrat', 'steward', 'checkuser', 'oversight' );
+$wgRemoveGroups['steward'] = array( 'sysop', 'bureaucrat', 'steward', 'checkuser', 'oversight' );
+$wgAddGroups['bureaucrat'] = array( 'bot', 'sysop', 'bureaucrat' );
+$wgRemoveGroups['bureaucrat'] = array( 'bot', 'sysop', 'bureaucrat' );
+$wgGroupsRemoveFromSelf = [ '*' => true ];
```
Comment (The-Voidwalker) [inline LocalSettings.php:374]: What is this even supposed to do?

Comment (bonnedav) [reply LocalSettings.php:374]: $wgGroupsRemoveFromSelf = [ '*' => true ];
will allow anyone to remove themselves (and only themselves) from any group they happen to be in. It basically enables self demotion.

**Comment by bonnedav** [comment]
@The-Voidwalker I think that removing redundant rights makes it more clear what exactly a group does and it's members can do that others can't. Say for instance I was a new user and saw that crat has block, before looking at sysop I might think that only crats can block and not sysops. I will however revert the deduplication if you wish.

File: LocalSettings.php (hunk 1 above)
Comment (The-Voidwalker) [reply LocalSettings.php:374]: Neat

**Comment by bonnedav** [comment]
@The-Voidwalker what is going on whit the reviewer and editor groups? i copied their grants directly from the flagged revs plugin but they are unchanged and still missing rights. it worked on my own wiki. maybe you should update the flagged revs plugin?

**Pull request merged**